x-pack/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/categorization_view/category_stopped_partitions.tsx

```diff
@@ -0,0 +1,141 @@
+import { i18n } from '@kbn/i18n';
+import { from } from 'rxjs';
+import { switchMap, takeWhile, tap } from 'rxjs/operators';
+import { JobCreatorContext } from '../../../job_creator_context';
+import { CategorizationJobCreator } from '../../../../../common/job_creator';
+import { ml } from '../../../../../../../services/ml_api_service';
+import { extractErrorProperties } from '../../../../../../../../../common/util/errors';
+
+const NUMBER_OF_PREVIEW = 5;
+export const CategoryStoppedPartitions: FC = () => {
+  const { jobCreator: jc, resultsLoader } = useContext(JobCreatorContext);
+  const jobCreator = jc as CategorizationJobCreator;
+  const [tableRow, setTableRow] = useState<Array<{ partitionName: string }>>([]);
+  const [stoppedPartitionsError, setStoppedPartitionsError] = useState<string | undefined>();
+
+  // using ref so this Subscription instance is only intialized once
+
+  const columns = useMemo(
+    () => [
+      {
+        field: 'partitionName',
+        name: i18n.translate(
+          'xpack.ml.newJob.wizard.pickFieldsStep.stoppedPartitionsPreviewColumnName',
+          {
+            defaultMessage: 'Stopped partition names',
+          }
+        ),
+        render: (partition: any) => (
+          <EuiText size="s">
+            <code>{partition}</code>
+          </EuiText>
+        ),
+      },
+    ],
+    []
+  );
+
+  const loadCategoryStoppedPartitions = useCallback(async () => {
+    try {
+      const results = await ml.results.getCategoryStoppedPartitions([jobCreator.jobId]);
+
+      if (
+        results?.jobs !== undefined &&
+        !Array.isArray(results?.jobs) && // if jobs is object of jobId: [partitions]
+        Array.isArray(results?.jobs[jobCreator.jobId]) &&
+        results.jobs[jobCreator.jobId].length > 0
+      ) {
+        return results.jobs[jobCreator.jobId];
+      }
+      return undefined;
+    } catch (e) {
+      const error = extractErrorProperties(e);
+      // might get 404 because job has not been created yet and that's ok
+      if (error.statusCode !== 404) {
+        setStoppedPartitionsError(error.message);
+      }
+    }
+  }, [jobCreator.jobId]);
+
+  useEffect(() => {
+    // only need to run this check if jobCreator.perPartitionStopOnWarn is turned on
+    if (jobCreator.perPartitionCategorization && jobCreator.perPartitionStopOnWarn) {
+      // subscribe to result updates
+      const resultsSubscription = resultsLoader.results$
+        .pipe(
+          switchMap(() => {
+            return from(loadCategoryStoppedPartitions());
+          }),
+          tap((results) => {
+            if (Array.isArray(results)) {
+              setTableRow(
+                results.slice(0, NUMBER_OF_PREVIEW).map((partitionName) => ({
+                  partitionName,
+                }))
+              );
+            }
+          }),
+          takeWhile((results) => {
+            return !results || (Array.isArray(results) && results.length <= NUMBER_OF_PREVIEW);
+          })
```

Comment:
When I shared a pseudo-code example I missed one point - `takeWhile` operator should be placed before `switchMap` where you perform stopped partitions fetching. With a current implementation potentially the stream can perform an unnecessary API call. It is not critical, but worth mentioning. 🙂 Can be improved later.